src/java/src/com/omniti/reconnoiter/event/NoitMetric.java

```diff
@@ -60,6 +60,15 @@ public NoitMetric(String[] parts, NoitMetricNumeric _nmn) throws Exception {
   }
   public NoitMetric(String[] parts) throws Exception {
     super(parts);
+
+    // Transform an MT-record back to a usual M-record for construction
+    if(parts[0].equals("MT")) {
+       String m_record[] = new String[7];
+       m_record[0] = "M";
+       for(int i=1;i<7;i++) { m_record[i] = parts[i]; }
```

Comment:
Is somewhere checked that parts.length is actually 7?
If now we'd also have the same problem at line 72: parts[5]